CoreGraphUtilities/src/au/gov/asd/tac/constellation/graph/utilities/hashmod/Hashmod.java

```diff
@@ -0,0 +1,171 @@
+        parser = new HashmodCSVImportFileParser();
+        try {
+            data = parser.parse(new HashmodInputSource(new File(csvFileStr)), null);
+        } catch (IOException ex) {
+            fail("IO Exception : " + ex.getLocalizedMessage());
+        }
+    }
+
+    public String[] getCSVFileHeaders() {
+        if (data != null && data.size() > 0) {
+            return data.get(0);
+        }
+        return null;
+    }
+
+    public String[] getCSVRow(final int row) {
+        if (data != null && data.size() > row) {
+            return data.get(row);
+        }
+        return null;
+    }
+
+    public String getCSVKey() {
+        String[] headers = getCSVFileHeaders();
+        if (headers != null && headers.length > 0) {
+            if (headers.length > 0) {
+                return headers[0];
+            }
+        }
+        return null;
+    }
+
+    public HashMap<String, Integer> getCSVKeys() {
+        final HashMap<String, Integer> keys = new HashMap<>();
+        if (data != null && data.size() > 0) {
+            int i;
+            for (i = 1; i < data.size(); i++) {
+                final String[] row = getCSVRow(i);
+                if (row[0] != null) {
+                    keys.put(row[0].toUpperCase(), 0);
+                }
+            }
+        }
+        return keys;
+    }
+
+    public int getNumberCSVColumns() {
+        String[] headers = getCSVFileHeaders();
+        if (headers != null) {
+            return headers.length;
+        }
+        return 0;
+    }
+
+    public String getCSVHeader(final int col) {
+        String[] headers = getCSVFileHeaders();
+        if (headers != null && headers.length > col) {
+            return headers[col];
+        }
+        return null;
+    }
+
+    public List<String[]> getCSVFileData() {
+        if (data != null && data.size() > 0) {
+            return data;
+        }
+        return null;
+    }
+
+    public String getValueFromKey(final String key, final int value) {
+        if (data != null && data.size() > 0) {
+            int i;
+            for (i = 1; i < data.size(); i++) {
+                String[] row = getCSVRow(i);
+                if (row[0].equalsIgnoreCase(key)) {
+
+                    if (row.length > value) {
+                        return row[value];
+                    }
+                }
+            }
+        }
+        return null;
+    }
+
+    public Boolean doesKeyExist(final String key) {
+        if (key == null) {
+            return false;
+        }
+        if (data != null && data.size() > 0) {
+            int i;
+            for (i = 1; i < data.size(); i++) {
+                String[] row = getCSVRow(i);
+                if (row[0].equalsIgnoreCase(key)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public String getCSVFileName() {
+        if (this.csvFileStr == null) {
+            return "";
+        }
+        return this.csvFileStr;
```

Comment:
more a personal preference but this can all be done in one line with the ternary operator:
`return this.csvFileStr == null ? "" : this.csvFileStr`